### PR TITLE
Chore/null setting

### PR DESCRIPTION
**Pull request by saltcod**
Confirm before setting table cell text value to null

screenshot-2024-06-21-at-11 59 26

screenshot-2024-06-21-at-11 59 57

### CODE REVIEW

File: apps/studio/components/grid/components/editor/TextEditor.tsx [hunk 1]
```diff
@@ -102,89 +104,104 @@ export const TextEditor = <TRow, TSummaryRow = unknown>({
...
+        confirmLabel="Confirm"
+        onCancel={() => setIsConfirmNextModalOpen(false)}
+        onConfirm={() => {
+          saveChanges(null)
+        }}
       >
-        {value === null ? <NullValue /> : value === '' ? <EmptyValue /> : value}
-      </div>
-    </Popover>
+        <p className="py-4 text-sm text-foreground-light">
```
Comment (MildTomato) [inline apps/studio/components/grid/components/editor/TextEditor.tsx:201]: you shouldn't need py classes

File: apps/studio/components/grid/components/editor/TextEditor.tsx [hunk 2]
```diff
@@ -102,89 +104,104 @@ export const TextEditor = <TRow, TSummaryRow = unknown>({
...
-              onChange={onChange}
-            />
-            {isEditable && (
-              <div className="flex items-start justify-between p-2 bg-surface-200 space-x-2">
-                <div className="space-y-1">
-                  <div className="flex items-center space-x-2">
-                    <div className="px-1.5 py-[2.5px] rounded bg-surface-300 border border-strong flex items-center justify-center">
-                      <span className="text-[10px]">⏎</span>
+    <>
+      <Popover
```
Comment (MildTomato) [inline apps/studio/components/grid/components/editor/TextEditor.tsx:108]: This looks deprecated

File: apps/studio/components/grid/components/editor/TextEditor.tsx [hunk 3]
```diff
@@ -102,89 +104,104 @@ export const TextEditor = <TRow, TSummaryRow = unknown>({
...
-                      size="tiny"
-                      onClick={() => saveChanges(null)}
-                    >
-                      <div>Set to NULL</div>
-                    </Button>
-                  )}
+                        size="tiny"
+                        onClick={() => setIsConfirmNextModalOpen(true)}
+                      >
+                        <div>Set to NULL</div>
```
[the remark this answers is not in the document]
Comment (joshenlim) [reply apps/studio/components/grid/components/editor/TextEditor.tsx:173]: i reckon we dont need div and just remove asChild on the button

File: apps/studio/components/grid/components/editor/TextEditor.tsx (hunk 2 above)
Comment (joshenlim) [reply apps/studio/components/grid/components/editor/TextEditor.tsx:108]: i'll refactor these in another PR

File: apps/studio/components/grid/components/editor/TextEditor.tsx [hunk 4]
```diff
@@ -102,89 +104,104 @@ export const TextEditor = <TRow, TSummaryRow = unknown>({
...
+        onCancel={() => setIsConfirmNextModalOpen(false)}
+        onConfirm={() => {
+          saveChanges(null)
+        }}
       >
-        {value === null ? <NullValue /> : value === '' ? <EmptyValue /> : value}
-      </div>
-    </Popover>
+        <p className="py-4 text-sm text-foreground-light">
+          Are you sure you wish to set this value to null? <br /> This can't be undone.
```
Comment (joshenlim) [inline apps/studio/components/grid/components/editor/TextEditor.tsx:202]: just gonna change can't -> cannot

**Comment by joshenlim** [comment]
Found some issues
- If you edit the field, then hit "Set to NULL", then hit "Cancel" -> the popover closes and the changes are saved
  - I think the right behaviour is that the user should still be in an editing state
- If you edit the field, then hit "Set to NULL", then hit "Confirm" -> the popover closes and the changes are saved (cell is not set to NULL)
  - Cell value should've been set to NULL and the popover closes

**Review by joshenlim — APPROVED** [review]
LGTM! Tested on the preview in the table editor for a text type column:
- Clicking set to NULL and hitting cancel
- Clicking set to NULL and hitting confirm

**Pull request merged**